### ISSUE TITLE
Confusing/false positive arguments-differ on enum _generate_next_value_

### ISSUE THREAD

**Issue by scop** [PyCQA/pylint#5371]
### Bug description

```python
# pylint: disable=missing-class-docstring,missing-module-docstring

from __future__ import annotations

from enum import Enum, auto
from typing import Any

class MyEnum(Enum):
    @staticmethod
    def _generate_next_value_(name: str, start: int, count: int, last_values: list[Any]) -> Any:
        return 42

class FooEnum(MyEnum):
    BAR = auto()

print(FooEnum.BAR)
```

### Command used

```shell
pylint t.py
```

### Pylint output

```shell
************* Module t
t.py:10:4: W0221: Number of parameters was 4 in 'Enum._generate_next_value_' and is now 4 in overridden 'MyEnum._generate_next_value_' method (arguments-differ)
```

### Expected behavior

No error. Or an error message that makes sense, if there is an error :) Currently it says the number is 4 for both which doesn't help with pinpointing the problem.

Somehow, it's the `@staticmethod` that throws pylint off here. That's however necessary for mypy: https://github.com/python/mypy/issues/7591

### Pylint version

```shell
pylint 2.11.1
astroid 2.9.0
Python 3.9.4 (default, Apr 11 2021, 07:48:09) 
[GCC 9.3.0]
```
